compiler/rustc_ast/src/ast.rs

```diff
@@ -1614,6 +1621,8 @@ pub enum ExprKind {
     /// A `format_args!()` expression.
     FormatArgs(P<FormatArgs>),
 
+    UnsafeBinderCast(UnsafeBinderCastKind, P<Expr>, Option<P<Ty>>),
```

Comment:
docs

Reply:
What's your reason for going with specifying the full type instead of a scheme that has to specify the lifetimes on the user-syntax side? The inability to specify most lifetimes inside bodies?

Reply:
> instead of a scheme that has to specify the lifetimes on the user-syntax side?

wdym by this?

Reply:
if you have an `unsafe<'x, 'y> &'x Thing<'y>` I expected something along the lines of `unwrap!(thing, 'a, 'static)` instead of what you have here: `unwrap!(thing, &'a Thing<'static>)`. Kind of instantiating the binder manually like when specifying generics.

Reply:
yea... I guess with the common case being actually not specifying it in the macro at all, but having it in a field type and storing a value there, it's not actually much of an issue to repeat a type.